homeassistant/components/systemmonitor/sensor.py

```diff
@@ -415,6 +416,25 @@ def check_legacy_resource(resource: str, resources: set[str]) -> bool:
     return False
 
 
+async def clean_obsolete_entities(
+    hass: HomeAssistant, entry_id: str, loaded_resources: set[str]
+) -> None:
```

Comment:
I think, we can make this a callback function, as there is nothing that needs to be awaited.

Reply:
Yeah. That's right.